plugins/inputs/github/github.go

```diff
@@ -0,0 +1,168 @@
+package github
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+	"sync"
+	"time"
+
+	gh "github.com/google/go-github/github"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"golang.org/x/oauth2"
+)
+
+// GitHub - plugin main structure
+type GitHub struct {
+	Repositories []string          `toml:"repositories"`
+	AccessToken  string            `toml:"access_token"`
+	HTTPTimeout  internal.Duration `toml:"http_timeout"`
+	githubClient *gh.Client
+}
+
+const sampleConfig = `
+  ## Specify a list of repositories
+  repositories = ["influxdata/influxdb"]
+
+  ## API Key for GitHub API requests
+  api_key = ""
+
+  ## Timeout for GitHub API requests
+  http_timeout = "5s"
+`
+
+// NewGitHub returns a new instance of the GitHub input plugin
+func NewGitHub() *GitHub {
+	return &GitHub{
+		HTTPTimeout: internal.Duration{Duration: time.Second * 5},
+	}
+}
+
+// SampleConfig returns sample configuration for this plugin.
+func (github *GitHub) SampleConfig() string {
+	return sampleConfig
+}
+
+// Description returns the plugin description.
+func (github *GitHub) Description() string {
+	return "Read repository information from GitHub, including forks, stars, and more."
+}
+
+// Create HTTP Client
+func (github *GitHub) createGitHubClient() (*gh.Client, error) {
```

Comment:
Go style is generally to use a single letter for the function receiver, I would probably do this and as a bonus you won't have to alias the github package.